[TW-118] Refresh "Building request workflows" (#3759)
postmanlabs/postman-docs: src/pages/docs/running-collections/building-workflows.md

---

Typically when you start a [collection run](/docs/running-collections/intro-to-collection-runs/), Postman runs all requests in the same order they appear in your collection. Requests in folders are executed first, followed by any requests in the root of the collection.

In the Collection Runner, you have the option to change the order of the requests before starting a run. However, instead of manually changing the request order each time you run the collection, you can automate this behavior using the `postman.setNextRequest()` function.

As the name suggests, `postman.setNextRequest()` enables you to specify which request Postman runs next, following the current request. Using this function, you can build custom workflows that chain requests, running them one after the other in a specific order.

<img alt="Setting the next request" src="https://assets.postman.com/postman-docs/set-next-request-v9-4.jpg" width="841px">

## Setting the next request

> To specify the request to run next, add the following code on the **Tests** tab of a request. For `request_name`, use the name of the request you want to run next.

Reviewer: Maybe: 
```suggestion
To specify the request to run next, add the following code on the **Tests** tab of a request. Replace `request_name` with the name of the request you want to run next.
```

```js
postman.setNextRequest("request_name");
```

Postman runs the specified request after completing the current request.

## Looping over a request

> If you pass the name of the current request to the `setNextRequest` function, Postman will run the current request repeatedly on a continuous loop.

Reviewer: I think this sentence could omit either "repeatedly" or "continuous"

<img alt="Looping over a request" src="https://assets.postman.com/postman-docs/set-next-request-loop-v9-4.jpg" width="841px">

> **IMPORTANT:** Make sure to wrap `setNextRequest` in some additional logic so the request doesn't loop indefinitely. For example, you might exit the loop after a certain number of iterations or when another condition is met. Otherwise you will need to force close the Collection Runner to end the loop.

## Stopping a workflow

To stop a workflow, add the following code on the **Tests** tab of a request.

```js
postman.setNextRequest(null);
```

The collection run will stop after Postman completes the current request.

## Tips for building request workflows

Keep the following tips in mind when using the `postman.setNextRequest()` function.

### setNextRequest() only works in Collection Runner

The `postman.setNextRequest()` function has no effect when you run a request using the **Send** button and is only used when you run a collection.

### Use setNextRequest() in pre-request or test scripts

> You can use `postman.setNextRequest()` in the pre-request script or the test script of a request. If more than one valued is assigned, the last value that is set takes precedence.

Reviewer: ```suggestion
You can use `postman.setNextRequest()` in the pre-request script or the test script of a request. If more than one value is assigned, the last value that is set takes precedence.
```
